demo/src/Components/BottomSheet/BottomSheet.tsx

```diff
@@ -111,23 +214,182 @@ const BottomSheet = (props: HvComponentProps) => {
     };
   }, [onHyperviewEventDispatch, visible]);
 
-  const children = Hyperview.renderChildren(
-    props.element,
+  const childNodes = Array.from(props.element.childNodes).filter(
+    n => (n as Element).tagName !== 'bottom-sheet:stop-point',
+  );
+  const children = Hyperview.renderChildNodes(
+    childNodes,
     props.stylesheets,
     props.onUpdate,
     props.options,
   );
 
+  const findBottomSheetEndPoint = () => {
+    'worklet';
+
+    const SWIPE_TO_CLOSE_THRESHOLD = 0.1;
+    const changeY = translateY.value + velocity.value * SCREEN_HEIGHT;
+
+    if (
+      hvProps.swipeToClose &&
+      -changeY < SCREEN_HEIGHT * SWIPE_TO_CLOSE_THRESHOLD
+    ) {
+      animateClose();
+      return;
+    }
+
+    if (contentSectionHeights.length > 0) {
+      let cumlHeight = 0;
+      contentSectionHeights.forEach((csHeight, index) => {
+        cumlHeight += csHeight;
+        stopPointLocations[index] = Math.min(cumlHeight / SCREEN_HEIGHT, 1.0);
+      });
+      console.log('stopPointLocations', stopPointLocations);
+    }
+
+    if (stopPointLocations.length > 0) {
```

Comment:
This code block is quite intricate, I think it would make sense to pull into helper(s) and add unit tests